src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/FunctionApprovalRequestContent.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Represents a request for user approval of a function call.
+/// </summary>
+[Experimental("MEAI001")]
+public sealed class FunctionApprovalRequestContent : UserInputRequestContent
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FunctionApprovalRequestContent"/> class.
+    /// </summary>
+    /// <param name="id">The ID to uniquely identify the function approval request/response pair.</param>
+    /// <param name="functionCall">The function call that requires user approval.</param>
+    public FunctionApprovalRequestContent(string id, FunctionCallContent functionCall)
+        : base(id)
```

Comment:
I think 99% of calls will be passing the FCC.CallId, it may be worth changing the ctor.
```suggestion
    public FunctionApprovalRequestContent(FunctionCallContent functionCall)
        : base(functionCall.CallId)
```